File: buddybridge-frontend/src/app/restrict/base/financeiro/movimento/model/movimentacao.ts
import { Classificacao } from "../../classificacao/model/classificacao";


export class Movimentacao {
  idMovimentacao?: number;
  historico?: string;
  dataLancamento?: Date;
  classificacao?: Classificacao | null;
  valor: number | null;
  observacoes?: string;
  valorPendente?: number | null;;

  constructor() {
    this.idMovimentacao = undefined;
    this.historico = '';
    this.dataLancamento = new Date();
    this.classificacao = new Classificacao();
    this.valor = 0;
    this.observacoes = '';
    this.valorPendente = 0;
  }
}

File: buddybridge-backend/src/main/java/br/com/buddybridge/core/financeiro/movimentacao/entity/Movimentacao.java
package br.com.buddybridge.core.financeiro.movimentacao.entity;

import br.com.buddybridge.core.financeiro.classificacao.entity.Classificacao;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDate;

@Entity
@Table(name = "movimentacao")
@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
public class Movimentacao {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_movimentacao")
    private Long idMovimentacao;

    @Column(name = "historico", nullable = false)
    private String historico;

    @Column(name = "data_lancamento", nullable = false)
    private LocalDate dataLancamento;

    @ManyToOne
    @JoinColumn(name = "id_classificacao", nullable = false)
    private Classificacao classificacao;

    @Column(name = "valor", nullable = false)
    private BigDecimal valor;

    @Column(name = "observacoes")
    private String observacoes;

    @Column(name = "valor_pendente", nullable = false)
    private BigDecimal valorPendente;
}
